### DISCUSSION TITLE
Fixing unscoped enum member conflicts via winmdgenerator?

### DISCUSSION THREAD

**Discussion by riverar** [microsoft/win32metadata#745] [Q&A] [answered]
Hey @sotteson1, the [DIA SDK metadata generation sample](https://github.com/microsoft/win32metadata/tree/master/sources/GeneratorSdk/samples/DiaSdk) emits a SymTagEnum enumeration with an identically named enumerator as such:

```cpp
enum SymTagEnum
{
// ...
    SymTagEnum,
// ...
};
```

As you can imagine, this poses some challenges with codegen.

I created a local `options.rsp` with the following contents:

```
--remap
SymTagEnum=SymTag
```

But this renames both the enumeration and the problematic enumerator.

Thoughts?

**Comment by riverar** [comment, marked as answer]
I manage to work around this with several files.

options.rsp
```
--remap
SymTagEnum=SymTag
--exclude
SymTagEnum
```

dia.manual.cs
```csharp
using Windows.Win32.Foundation;
using Windows.Win32.Interop;

namespace Microsoft.Dia
{
    public enum SymTag
    {
      // ...
    }
}
```

Am still interested in your thoughts. Thanks!

**Reply by sotteson1** [reply]
I don't know of a better way. I haven't seen an enum to that before. That's a pretty bad job of naming the original enum and value IMO!

**Reply by riverar** [reply]
Thanks for the sanity check! After I discovered the generator looked for `*.rsp`/`*.manual.cs` on disk, the world opened up to me. 😈
